pkg/controller/metadata/metadata_controller_test.go

```diff
@@ -558,3 +558,48 @@ func generateCertChain(t *testing.T, ctx context.Context, c client.Client, g *Go
 
 	return rootCertReq, intCertReq, metaCertReq, tearDown
 }
+
+func TestShouldRegenerate(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	const ConstantHash = "Im a constant hash"
+
+	mockSecrets := corev1.Secret{}
+	mockSecrets.ObjectMeta.Annotations = make(map[string]string)
+	mockSecrets.Data = make(map[string][]byte)
+
+	// Hashes should differ, so should be true to regenerate
+	mockSecrets.ObjectMeta.Annotations[versionAnnotation] = ""
+	g.Eventually(ShouldRegenerate(&mockSecrets, ConstantHash)).Should(BeTrue())
+
+	// Hash should now match, but there is no data for the expiration, this simulates a upgrade.
+	mockSecrets.ObjectMeta.Annotations[versionAnnotation] = ConstantHash
+	g.Eventually(ShouldRegenerate(&mockSecrets, ConstantHash)).Should(BeTrue())
+
+	// There should be a parse error.
+	mockSecrets.Data[validityDays] = []byte("30")
+	mockSecrets.Data[validUntil] = []byte("")
+	g.Eventually(ShouldRegenerate(&mockSecrets, ConstantHash)).Should(BeTrue())
+
+	// Should regenerate as time is in the past.
+	mockSecrets.Data[validUntil] = []byte("Mon, 02 Jan 2006 15:04:05 -0700")
+	g.Eventually(ShouldRegenerate(&mockSecrets, ConstantHash)).Should(BeTrue())
+
+	// Should regenerate if half of the validity days.
+	mockSecrets.Data[validUntil] = []byte(time.Now().AddDate(0, 0, 14).Format(time.RFC1123Z))
+	g.Eventually(ShouldRegenerate(&mockSecrets, ConstantHash)).Should(BeTrue())
+
+	// Should'nt regenerate as in the future.
+	mockSecrets.Data[validUntil] = []byte(time.Now().AddDate(0, 0, 60).Format(time.RFC1123Z))
+	g.Eventually(ShouldRegenerate(&mockSecrets, ConstantHash)).Should(BeFalse())
+
+	mockSecrets.Data[validUntil] = []byte(time.Now().AddDate(0, 0, 1120).Format(time.RFC1123Z))
```

Comment:
https://github.com/alphagov/verify-metadata-controller/blob/d805d91cfb17541fe7912ff97b4866584c2cf9f3/pkg/controller/metadata/metadata_controller_test.go#L596-L597

This test is unnecessary. I know you added it after my previous comment but I didn't notice that the `validityDays` is set to 30 days above. So the previous test with `AddDate(0, 0, 60)` is enough to test the case in the future.